pkg/server/api/entry/v1/service.go

```diff
@@ -169,6 +175,83 @@ func (s *Service) BatchDeleteEntry(ctx context.Context, req *entry.BatchDeleteEn
 	}, nil
 }
 
+func (s *Service) updateEntry(ctx context.Context, e *types.Entry, inputMask *types.EntryMask, outputMask *types.EntryMask) *entry.BatchUpdateEntryResponse_Result {
+	log := rpccontext.Logger(ctx)
+
+	if e == nil {
+		log.Error("missing entry")
+		return &entry.BatchUpdateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "missing entry"),
+		}
+	}
+
+	if inputMask == nil {
+		log.Error("missing input mask")
+		return &entry.BatchUpdateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "missing input mask"),
+		}
+	}
+	if e.SpiffeId != nil {
+		log = log.WithField(telemetry.SPIFFEID, e.SpiffeId)
+	}
+	// Fetch the previous version of the entry.
+	res, err := s.ds.FetchRegistrationEntry(ctx,
+		&datastore.FetchRegistrationEntryRequest{EntryId: e.Id},
+	)
+	if err != nil {
+		log.WithError(err).Error("failed to fetch entry")
+		return &entry.BatchUpdateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "failed to fetch entry"),
+		}
+	}
+	oldEntry := res.Entry
+
+	// In general it's illegal for an entry to have an empty ParentId or SpiffeId.
+	// But, when updating, it's OK for these fields to be empty as long as they aren't being changed.
+	// To prevent an error we put dummy data in the update request (the actual values don't matter).
```

Comment:
To avoid this, we could not use ProtoToRegistrationEntry and instead have code to do this that is specific to updating. It seemed simpler to put in a dummy SpiffeID when needed even thought this is a bit ugly.